### DISCUSSION TITLE
`createContractEventFilter` returns no events without setting a fromBlock

### DISCUSSION THREAD

**Discussion by DannyDelott** [wevm/viem#947] [Question]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

1.4.2

### Current Behavior

The `fromBlock` arg is typed as optional in `createContractFilter`, but you won't get any events back without it. 

![image](https://github.com/wagmi-dev/viem/assets/4524175/a0942c99-922a-4906-88e4-b94a3b7cb768)

### Expected Behavior

If you don't pass a `fromBlock`, it should default to `"earliest"`. At a minimum, this keeps the mental model the same between viem and ethers. (see: https://docs.ethers.org/v6/api/contract/#BaseContract-queryFilter)

### Steps To Reproduce

See repro example

### Link to Minimal Reproducible Example (StackBlitz, CodeSandbox, GitHub repo etc.)

https://stackblitz.com/edit/viem-getting-started-3nq49j?file=index.ts,erc20.ts,index.html

### Anything else?

![image](https://github.com/wagmi-dev/viem/assets/4524175/1bbc3536-2d26-4d8e-83dc-ed699e069b47)

**Comment by jxom** [comment]
Isn’t this expected? I’d expect for it to create a filter at the latest block.

Changing this would be a breaking change, and I would rather have the behaviour in sync with the node.
